src/coreclr/jit/codegenriscv64.cpp

```diff
@@ -3374,8 +3374,17 @@ void CodeGen::genFloatToIntCast(GenTree* treeNode)
     }
 
     genConsumeOperands(treeNode->AsOp());
+    regNumber tmpReg = rsGetRsvdReg();
 
     GetEmitter()->emitIns_R_R(ins, EA_8BYTE, treeNode->GetRegNum(), op1->GetRegNum());
+
+    // This part emulates the "flush to zero" option because the RISC-V specification does not provide it.
+    unsigned mask = 16; // NV = 1 | DZ = 0 | OF = 0 | UF = 0 | NX = 0
+    GetEmitter()->emitIns_R_I_I(INS_csrrci, EA_8BYTE, tmpReg, mask, 1/*fflags*/);
```

Comment:
Could you share any document which NaN input set the invalid operation exception flag in `fcvt_*`? Actually, I cannot find. Thank you.

Reply:
Yes, unprivilege specification - page 69 - "Single-Precision Floating-Point Conversion and Move
Instructions"
https://drive.google.com/file/d/1s0lZxUZaa7eV_O0_WsZzaurFLLww7ou5/view?usp=drive_link
"If the rounded result is not representable in the destination format, it
is clipped to the nearest value and the invalid flag is set."
Thus not only NaN input, but any input that can't be represented in target type sets the invalid flag.